toolchain/check/name_component.cpp

```diff
@@ -11,76 +11,63 @@ namespace Carbon::Check {
 
 auto PopNameComponent(Context& context, SemIR::InstId return_slot_pattern_id)
     -> NameComponent {
-  Parse::NodeId first_param_node_id = Parse::InvalidNodeId();
-  Parse::NodeId last_param_node_id = Parse::InvalidNodeId();
+  NameComponent name_component = {
+      .name_loc_id = Parse::NodeId::Invalid,
+      .name_id = SemIR::NameId::Invalid,
+      .first_param_node_id = Parse::NodeId::Invalid,
+      .last_param_node_id = Parse::NodeId::Invalid,
+      .implicit_params_loc_id = Parse::NodeId::Invalid,
+      .implicit_param_patterns_id = SemIR::InstBlockId::Invalid,
+      .params_loc_id = Parse::NodeId::Invalid,
+      .param_patterns_id = SemIR::InstBlockId::Invalid,
+      .call_params_id = SemIR::InstBlockId::Invalid,
+      .return_slot_pattern_id = return_slot_pattern_id,
+      .pattern_block_id = SemIR::InstBlockId::Invalid,
+  };
```

Comment:
Previously, I could verify each field was being set on construction. What made you choose to switch to this invalid initialization?